code/game/machinery/pipe/construction.dm

```diff
@@ -201,6 +194,83 @@ Buildable meters
 
 	qdel(src)
 
+/**
+ * Attempt to automatically resolve a pipe conflict by reconfiguring any smart pipes involved.
+ *
+ * Constraints:
+ *  - A smart pipe cannot have current connections reconfigured.
+ *  - A smart pipe cannot have fewer than two directions in which it will connect.
+ *  - A smart pipe, existing or new, will not automatically reconfigure itself to permit directions it was not previously permitting.
+ */
+/obj/item/pipe/proc/try_smart_reconfiguration(obj/machinery/atmospherics/machine, our_init_dirs, mob/living/user)
+	// If we're a smart pipe, we might be able to solve this by placing down a more constrained version of ourselves.
+	var/obj/machinery/atmospherics/pipe/smart/other_smart_pipe = machine
+	if(ispath(pipe_type, /obj/machinery/atmospherics/pipe/smart/))
+		// If we're conflicting with another smart pipe, see if we can negotiate.
+		if(istype(other_smart_pipe))
+			// Two smart pipes. This is going to get complicated.
+			// Check to see whether the already placed pipe is bent or not.
+			if (ISDIAGONALDIR(other_smart_pipe.dir))
+				// The other pipe is bent, with at least two current connections. See if we can bounce off it as a bent pipe in the other direction.
```

Comment:
No complaints, I just really like that you've implemented this bit of logic it's been bugging me for a bit